### ISSUE TITLE
Display platform numbers (ref=* of railway=platform)

### ISSUE THREAD

**Issue by SelfishSeahorse** [gravitystorm/openstreetmap-carto#3271]
I think it would be quite useful if platform numbers, i.e. `ref=*` of `railway=platform`, were rendered.

**Comment by kocio-pl** [comment]
We're currently rendering name, so how do you think it should be rendered, ref instead of name or in any other way?

**Comment by SelfishSeahorse** [comment]
I'd say ref instead of name because the name of the station is already rendered.

**Comment by boothym** [comment]
Suppose it's a question of whether platforms should be displayed as "Platform X" like here: https://www.openstreetmap.org/#map=19/55.95148/-3.19063 (using name=) 
or with just their platform number like here: https://www.openstreetmap.org/#map=19/51.52932/-0.13498

I'd say it's better with simply their number, in which case osm-carto should render ref instead of name - it would also prevent people tagging for the renderer currently using the name tag.

**Comment by kocio-pl** [comment]
Sounds reasonable for me. Would any of you like to prepare the code?

**Comment by SelfishSeahorse** [comment]
I'd like to help, but unfortunately I have only little programming knowledge. :-(

**Comment by kocio-pl** [comment]
Me too, so that should not be a problem in many cases, it's rather pattern copying and slightly changing them. 

The problem is we lack coders in this project, so until somebody actually do it, there's little chance to resolve it.

**Comment by Adamant36** [comment]
Theres a way text from multiple fields can be displayed in labels with brackets I was just reading about. Maybe that would be an option. The code for it and an example is about half day down in the link.  

https://www.mapbox.com/help/studio-classic-cartocss/

**Comment by jragusa** [comment]
In France, Germany and Switzerland, ref platform are displayed white in a blue square. We could use the blue square of railway station as a shield like for ref label of road. Once restrained to z19 (z18?) this would not disturb the map and this allows to add the name tag if the latter is different.

**Comment by boothym** [comment]
I had a look on overpass-turbo for platforms with names in the UK and noticed a couple of things. 

Ways with name=* are not rendered, whereas closed ways with names (and area=yes) are rendered. 
**Is it possible to render ref= on single lines as well as closed ways?**

Most platforms which have ref=and name= can be described as follows:

- ref=1 and name=1
- ref=1 and name=Platform 1
- ref= and name=1 (tagging for the renderer)
- name=<name of station>
- name=Jubilee Eastbound (on London Underground stations, however most also have subway=yes so these are not be rendered)

So in terms of tagging, should names even be used if they are mostly just duplicating the ref?

**Comment by DaveF63** [comment]
My two pennies' worth:
Render `ref` not `name`
Delete `name ` as it's duplicating information which leads to confusion & errors.
Don't use "Platform", just a number. I see it as superfluous. It's similar to railway station names where we don't suffix "rail station" to the end of the name.

**Comment by boothym** [comment]
@kocio-pl I searched the code for "platform" and also looked at other places mentioning "railway", but can't find where the code is which actually renders the name. There is https://github.com/gravitystorm/openstreetmap-carto/blob/1419ef8e0dcdb0863207aedd6a93b5b8e26edfa2/roads.mss#L2132 but it's just the rendering of the area/line. Any ideas?

Plus there's the issue of why name=* renders on an area, but not on a single line.

**Comment by da1910** [comment]
Platform text rendering is handled as a roads-area-text-name, which only renders the name on an area over 3000px. To separate this out without retagging requires an exception, but at the moment doing so will remove labels where they are tagged with name.
I would support rendering the ref and ignoring the name tag, which I can implement, but it will temporarily remove labels in some cases, like Edinburgh linked above.

**Comment by boothym** [comment]
I don't think it will take that long to retag the platforms in OSM from name to ref - it shouldn't put you off switching the rendering from name to ref.

Is there a way to make a platform like this - without area=yes - render like it has area=yes?
https://www.openstreetmap.org/way/208763467

**Comment by da1910** [comment]
I don't see why not, but I'm not sure that that's really the best way to tag platforms? Surely that should be marked as area=yes since it's an actual platform?

**Comment by boothym** [comment]
Hmm you're probably right - and it would be better if an island platform was mapped as two separate ways so the ref tags can be added to each platform.

It would be great if were able to do a test render for this, and submit a PR :) Then we can start fixing the platforms in OSM by moving data from name to ref!

**Comment by boothym** [comment]
Looking again it seems that name=* is not rendered unless the railway=platform is also tagged as area=yes. That's fine for detailed stations where platforms are mapped as a closed way, but not for those with just a line for a platform.

So as well as switching from rendering name to ref, we also need to allow single ways to be rendered with the ref value.

**Comment by DaveF63** [comment]
The rules for railway=platform should be the same as for golf=hole where the 'name' tag is rendered if present, otherwise the 'ref' tag is displayed (if present).
